go.mod

```diff
@@ -34,14 +35,13 @@ require (
 	github.com/spf13/pflag v1.0.3
 	github.com/spf13/viper v1.3.1
 	github.com/stretchr/testify v1.3.0
-	go.opencensus.io v0.18.0
-	golang.org/x/net v0.0.0-20190311183353-d8887717615a
-	golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4 // indirect
+	go.opencensus.io v0.22.0
+	golang.org/x/net v0.0.0-20190501004415-9ce7a6920f09
 	golang.org/x/time v0.0.0-20180412165947-fbb02b2291d2
 	golang.org/x/tools v0.0.0-20190328211700-ab21143f2384
 	google.golang.org/api v0.0.0-20190117000611-43037ff31f69 // indirect
-	google.golang.org/genproto v0.0.0-20190111180523-db91494dd46c
-	google.golang.org/grpc v1.17.0
+	google.golang.org/genproto v0.0.0-20190425155659-357c62f0e4bb
+	google.golang.org/grpc v1.20.1
```

Comment:
grpc bump and protobuf bump. @markmandel is that ok ? does it requires tooling update ?

Reply:
Good catch. That's a big jump. If we do this, yes - we should regenerate things, and confirm 

Do we need this update to update opencensus as well? Are they linked dependencies?

I'm almost wondering if this should be split into some separate PRs to make this easier to review?
1. The update to OpenCensus
2. The updates to the test code
3. Anything else?

Would that be possible?

Chrome actually crashes on this page :disappointed: I should fire up Firefox, it usually works.

I'll also dig through this this week too.

Reply:
I suspect they are linked dependencies. The only change I had made was . to upgrade to 0.22. The rest came from running go mod vendor and go mod tiny.

I cannot split the test changes. One consequence of moving to 0.22 is that @Kuqd 's hack of unregistering/registering no longer works reliably.

Reply:
I'll would like to have first a grpc/protobuf bump to the version you need.

Reply:
That seems like a good strategy to me :+1: grpc is annoyingly backward compatible, so i expect nothing will break even if we update only one side.

Reply:
FWIW, if grpc or genproto are not backwards compatible than a lot of things would break. Both libraries need to maintain backwards compatibility to be useful since clients and servers cannot be updated at the same time.

Reply:
So I would agree with @Kuqd - I think the right step would be:

1. Update gRPC to the version required here (one PR)
1. Then (another PR) one of us can regenerate the SDK toolchain, and make sure that all work appropriately.
1. Then we can apply the opencensus update and changes (another PR)

That should make it a nicer, and safer process. WDYT?

Reply:
Sure thing.